File: 백준/Bronze/9506. 약수들의 합/약수들의 합.py
while 1:
    a=int(input())
    if a==-1:
        break
    else:
        d,num=1,[]
        while d<a:
            if a%d==0:
                num+=[d]
            d+=1
    if sum(num)==a:
        print(a,"= ",end='')
        for i in range(len(num)):
            if(i==len(num)-1):
                print(num[i])
            else:
                print(num[i],"+ ",end='')
    else:
        print(a,"is NOT perfect.")

File: 백준/Bronze/9506. 약수들의 합/약수들의 합.cc
#include <iostream>
#include <numeric>
#include <vector>
using namespace std;

int main(void) {
    int a, d, s;
    vector<int> num;
    while (1) {
        cin >> a;
        if (a == -1)
            break;
        else {
            d = 1, num = {};
            while (d < a) {
                if (a % d == 0)
                    num.push_back(d);
                d += 1;
            }
        }
        s = accumulate(num.begin(), num.end(), 0);
        if (s == a) {
            cout << a << " = ";
            for (int i = 0; i < num.size(); i++) {
                if (i == num.size() - 1)
                    cout << num[i] << endl;
                else
                    cout << num[i] << " + ";
            }
        } else {
            cout << a << " is NOT perfect." << endl;
        }
    }
    return 0;
}